fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -1098,6 +1102,32 @@ protected DruidWebService buildMetadataDruidWebService(ObjectMapper mapper) {
         return buildDruidWebService(DruidClientConfigHelper.getMetadataServiceConfig(), mapper);
     }
 
+    /**
+     * Create a ResponseFormatResolver for Servlet objects.
+     * <p>
+     * Currently default types are json, jsonapi and csv types.
+     *
+     * @return A ResponseFormatResolver
+     */
+    protected ResponseFormatResolver buildResponseFormatResolver() {
+        return (format, containerRequestContext) -> {
+            Map<String, String> formatsMap = new LinkedHashMap<>();
+            formatsMap.put("application/json", "json");
+            formatsMap.put("application/vnd.api+json", "jsonapi");
+            formatsMap.put("text/csv", "csv");
+
+            String headerFormat = containerRequestContext.getHeaderString("Accept");
+            if (format != null || headerFormat == null) {
+                return format;
```

Comment:
If format != null, does it matter which value is passed in?

Also, if there is a null format there should still be a default value, in the case of the standard fili, it's json.

Reply:
I think the URI format should precedence header format right? So if URI format is not null, we will just return the URI format. Is this what you are asking about? 

As for your second question, null case is already handled in [FiliResponseWriterSelector](https://github.com/yahoo/fili/blob/master/fili-core/src/main/java/com/yahoo/bard/webservice/web/FiliResponseWriterSelector.java#L44)